LibraryManagement.cpp

```diff
@@ -167,61 +172,91 @@ int choose;
+                		bookMark.viewAllBooks();
                 		break;
             		case 2:
-                		quit=true;
+                		shouldLeaveApp=true;
                 		break;
             		default:
                 		cout<<"Invalid input.\n";
                 		break;
                 }
                 }
-
 		}
-
-	 else {
+		else {
 		cout<<"Invalid input."<<endl;
-	 }
+	 	}
 		return 0;
+		}
+		
+void Bookmark::registerNewUser() {
+	string username,password;
+    cout << "Enter username: ";
+    cin >> username;
+    cout << "Enter password: ";
+    cin >> password;
+
+    if (bookMark.registerUser(username, password)) {
+        cout << "You are successfully registered" << endl;
+        
+    } else {
+        cout << "Error: username already exists." << endl;
+        exit(0);
+    }
+}
 
-	}
-
-
-
-
+		
+void Bookmark::loginUser() {
+    string username, password;
+    do {
+        cout << "Enter username: ";
+        cin >> username;
+        cout << "Enter password: ";
+        cin >> password;
+        if (bookMark.isAlreadyRegistered(username, password)) {
+            cout << "Login successful" << endl;
+            break;
+        } else {
+            cout << "Incorrect username or password!" << endl;
+            cout << "Press 'y' to try again: ";
+            char tryAgain;
+            cin >> tryAgain;
+            tryAgain = tolower(tryAgain);
+            if (tryAgain != 'y') {
+                exit(0);
+                break;
+            }
+        }
+    } while (true);
+}
+		
 
-void Bookmark::addUser(int userCounter) {
+	
+void Bookmark::addUser() {
     string firstName, lastName, ID;
-    if(userCounter < 10) {
-		cout << "ADD USER\n\n";
-    	cout << "Enter first name: ";
-    	cin >> firstName;
-    	cout << "Enter last name: ";
-    	cin >> lastName;
-    while(ID.length() != 13) {
+    if(userCounter > 20) { 
+    	cout<<"Max users reached...\n";
+    	return;
+    }
+        cout << "ADD USER\n\n";
+        cout << "Enter first name: ";
+        cin >> firstName;
+        cout << "Enter last name: ";
+        cin >> lastName;
+        while(ID.length() != 13) {
             cout<<"Enter ID: ";
             cin>>ID;
-        if(ID.length() != 13) {
-            cout<<"ID must have 13 digits"<<endl;
+            if(ID.length() != 13) {
+                cout<<"ID must have 13 digits"<<endl;
+            }
         }
-	}
-
-    bookMark.users[bookMark.userCounter].setFirstName(firstName);
-    bookMark.users[bookMark.userCounter].setLastName(lastName);
-    bookMark.users[bookMark.userCounter].setID(ID);
-    bookMark.incrementUser(bookMark.userCounter);
-        cout << "\nUser added successfully."<<endl;
-        main();
-
-}
-	else {
-		cout<<"You have reached the maximum number of users that can be added!"<<endl;
-		main();
-	}
-}
-
-void Bookmark::editUser(int userCounter) {
+        bookMark.users[userCounter].updateUser(firstName, lastName, ID);
+		bookMark.incrementUser();
```

Comment:
why do you increment user counter if you edit the current user?

Reply:
incrementUser() is only used with the addUser function, the editUser() method does not have an incrementUser().